### PR TITLE
Optimizing unit tests

### PR DESCRIPTION
**Pull request by husseincoder**
I'm trying to contribute to this repo and I have already used it at work to monitor our backend application .. so I've decided to start looking at the unit tests to understand more about how the whole lib work and found some optimizations that could be added to some of the unit tests

### CODE REVIEW

File: metrics-core/src/test/java/com/codahale/metrics/CsvReporterTest.java [hunk 1]
```diff
@@ -175,21 +176,7 @@ private String csv(String... lines) {
...
-            while (bufferedReader.read(buf) != -1) {
-                buf.flip();
-                builder.append(buf);
-                buf.clear();
-            }
-        } finally {
-            input.close();
-        }
-        return builder.toString();
+        return new String(Files.readAllBytes(new File(dataDirectory, filename).toPath()));
```
Comment (arteam) [inline metrics-core/src/test/java/com/codahale/metrics/CsvReporterTest.java]: Nice

File: metrics-core/src/test/java/com/codahale/metrics/DerivativeGaugeTest.java [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
 
+import org.junit.Before;
 import org.junit.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class DerivativeGaugeTest {
-    private final Gauge<String> gauge1 = new Gauge<String>() {
```
Comment (arteam) [inline metrics-core/src/test/java/com/codahale/metrics/DerivativeGaugeTest.java]: I don't think there is a big improvement in replacing a simple gauge with a mock object. I would avoid mocking unless needed. Could you please revert this change?

Comment (husseincoder) [inline metrics-core/src/test/java/com/codahale/metrics/DerivativeGaugeTest.java]: Done

**Comment by arteam** [comment]
Thanks!
